### PR TITLE
Clean up in Yii2 module

### PR DESCRIPTION
**Pull request by SamMousa**
- No more static variables
- No more database connection caching (that didn't actually work)
- Support for multiple database connections / database services with names other than 'db'

This needs more testing; I've tested it in a few of my production apps.

Also somewhere in the docs we should update the "contract" about application state:

1. When a test starts you get a fresh application.
2. Just before a request is executed the `Request` and `Response` component are reloaded.
3. Application state is persisted after the request and reused for the next request (as long as it is not in the `Request` or `Response` component.
4. All PDO based connections are supported as long as they are subclasses of `yii\db\Connection` and fire the `AFTER_OPEN` event.

Database handling (for transactions)

1. All caches are cleared at the end of a test.
2. At the start of a test we register an event handler for `AFTER_OPEN`.
3. When the event is fired we generate a cache key based on connection settings.
4. We store the PDO object in the cache.
5. We store an entry in the dsn cache, this is used to identify unsupported cases (think connecting to the same database with different credentials, in which case the transaction creates 2 world views which could cause issues).
6. If we detect a duplicate PDO object we replace the new PDO with the existing one, this causes the new connection to be closed immediately and guarantees us that the new `Connection` object uses the same PDO object and thus has the same world view.

### CODE REVIEW

**Comment by SamMousa** [comment]
I think the failing tests are not caused by my changes.

File: src/Codeception/Lib/Connector/Yii2.php [hunk 1]
```diff
@@ -21,71 +24,43 @@ class Yii2 extends Client
...
-    /**
-     * @var \yii\db\Connection
-     */
-    public static $db; // remember the db instance
-
-    /**
-     * @var TestMailer
-     */
-    public static $mailer;
+//    public $defaultServerVars = [];
```
Comment (samdark) [inline src/Codeception/Lib/Connector/Yii2.php]: Should not be left commented out.

File: src/Codeception/Module/Yii2.php [hunk 2]
```diff
@@ -122,67 +127,124 @@
  * Maintainer: **samdark**
  * Stability: **stable**
  *
+// * @property \Codeception\Lib\Connector\Yii2 $client
```
Comment (samdark) [inline src/Codeception/Module/Yii2.php]: ?

Comment (SamMousa) [reply src/Codeception/Module/Yii2.php]: Commented that out to have my IDE track down the original definition.. shouldn't have been committed

**Comment by SamMousa** [comment]
Okay this works correctly now. @DavertMik is a runtime of 24 minutes (versus the 17 minutes you get when using just 5 tasks) acceptable?

As I said in our chat; if the extra time is not acceptable I'll reorganize it to test all frameworks in 1 job; I think the split approach gives clearer results though. Note that regardless of it being 17 or 24 minutes, I'm not going to wait for it anyway and this way I at least get preliminary results < 5 minutes.
We can make it even faster by excluding some more jobs that are not needed.

**Comment by DavertMik** [comment]
@SamMousa Wow, you are awesome. I don't really care of runtime but you did an amazing job in splitting tests between different jobs. I really like new `.travis.yml` :+1:

**Comment by DavertMik** [comment]
Could you push updated `.travis.yml` into another PR? I don't wan that Yii2 changes would be mixed up with travis settings history.

Also, if you think Yii cleanup is ready I can merge it

**Comment by SamMousa** [comment]
Will do!

**Comment by SamMousa** [comment]
Travis PR.
https://github.com/Codeception/Codeception/pull/4893

Don't merge this PR, I'll create a new one after the travis changes have been merged; that'll only contain the actual commits that are needed.

**Pull request closed without merging**